packages/xo-web/src/common/editable/index.js

```diff
@@ -294,18 +294,22 @@ export class Number extends Component {
     return +this.refs.input.value
   }
 
-  _onChange = (value, params) => {
-    if (value === '') {
-      if (this.props.nullable) {
-        value = null
+  _onChange = async (value, params) => {
+    try {
+      if (value === '') {
+        if (this.props.nullable) {
+          value = null
+        } else {
+          return
+        }
       } else {
-        return
+        value = +value
       }
-    } else {
-      value = +value
-    }
 
-    this.props.onChange(value, params)
+      await this.props.onChange(value, params)
+    } catch (error) {
+      throw error
```

Comment:
I don't understand the point of catching an error just to rethrow it.

Reply:
Indeed, the import change appears to be the `await`.

Reply:
> I don't understand the point of catching an error just to rethrow it.

The current behaviour is: if we edit a number and it's failing, the UI are not displaying an error, and for the user it's look like it worked.
Eg: If you try to edit a `rate limit network` as viewer, the modification will fail, and a success tooltip will appear next to the input

By throwing an error in the case of failure, the error [tooltip](https://github.com/vatesfr/xen-orchestra/blob/1ff25943dcedda9448cb56302adc696dc740efec/packages/xo-web/src/common/editable/index.js#L155) is displayed

Reply:
@MathieuRA Your comment does not explain why the `try`/`catch` is necessary.

Because what you're doing right now is catch an exception and then rethrow it, which is the same as not catching it in the first place.

Reply:
Ho okay, i understand.

So only the `await` is necessary